### PR TITLE
Add tests proving bug with ServerRequestParameters.populateQueryParams when passed null

### PR DESCRIPTION
**Pull request by jasonnutter**
Pulls in the fix from #1531 and adds tests to prove it is fixed.

### CODE REVIEW

File: lib/msal-core/src/ServerRequestParameters.ts [hunk 1]
```diff
@@ -82,7 +82,7 @@ export class ServerRequestParameters {
      * @param request
      * @param serverAuthenticationRequest
      */
-    populateQueryParams(account: Account, request: AuthenticationParameters, adalIdTokenObject?: object, silentCall?: boolean): void {
+    populateQueryParams(account: Account, request: AuthenticationParameters|null, adalIdTokenObject?: object, silentCall?: boolean): void {
```
Comment (tnorling) [inline lib/msal-core/src/ServerRequestParameters.ts:85]: I don't think we need to type things with ```|null``` since we don't have ```strictNullChecks``` enabled right? Seems like something we should get around to since it keeps coming up, but should enabling this and fixing the types be its own PR?

Comment (jasonnutter) [reply lib/msal-core/src/ServerRequestParameters.ts:85]: Yeah, you're right, but I was fine with it since we know want to do it and it doesn't cause any harm.

**Review by tnorling — APPROVED** [review]

**Review by pkanher617 — APPROVED** [review]
